### PR TITLE
Send events to event gate

### PR DESCRIPTION
**Pull request by stephanebisson**
Phabricator Link: 
- https://phabricator.wikimedia.org/T267345
- https://phabricator.wikimedia.org/T267344
- https://phabricator.wikimedia.org/T267346

### Problem Statement

We have to move away from the old Event logging system

### Solution

Send all events from the KaiOS app to the new event gate system. This applies to all 3 schemas: InukaPageView, FirstRun and Feedback.

### CODE REVIEW

File: src/utils/sendEvent.js [hunk 1]
```diff
@@ -1,22 +1,12 @@
...
-    event,
-    webHost: `${language}.wikipedia.org`,
-    wiki: `${language}wiki`
-  })
-  if (isUrlValid(url)) {
-    navigator.sendBeacon(url)
-  }
+export const sendEvent = ($schema, stream, event) => {
+  const body = JSON.stringify([{
+    $schema,
```
Comment (medied) [inline src/utils/sendEvent.js:7]: I gather the `$` is just a naming convention for the schema?

https://wikitech.wikimedia.org/wiki/Event_Platform/EventLogging_legacy#produce_event_data

File: src/hooks/useTracking.js [hunk 2]
```diff
@@ -100,7 +100,7 @@ export const useTracking = (
       /* eslint-enable camelcase */
     }
 
-    sendEvent(SCHEMA_NAME, SCHEMA_REV, language, event)
+    sendEvent(SCHEMA_NAME, STREAM_NAME, event)
```
Comment (medied) [inline src/hooks/useTracking.js:103]: Shall we remove `language` as an argument (line 28)?

Comment (stephanebisson) [reply src/hooks/useTracking.js:102]: Done

File: src/utils/consent.js [hunk 3]
```diff
@@ -8,10 +8,12 @@ export const isConsentGranted = () => {
...
-  sendEvent('KaiOSAppFirstRun', 20158341, getAppLanguage(), {
-    app_version: version,
-    app_id: appInstallId()
-  })
+  sendEvent(
+    '/analytics/legacy/kaiosappfirstrun/1.0.0',
+    'eventlogging_KaiOSAppFirstRun', {
+      app_version: version,
+      app_id: appInstallId()
+    })
```
Comment (hueitan) [inline src/utils/consent.js:16]:
```suggestion
    'eventlogging_KaiOSAppFirstRun', 
    {
      app_version: version,
      app_id: appInstallId()
    }
)
```

**Review by hueitan — CHANGES_REQUESTED** [review]
Current article preview tracking sent is response error 500, should this remove or remain old beacon link?

**Comment by stephanebisson** [comment]
> Current article preview tracking sent is response error 500, should this remove or remain old beacon link?

Good catch. VirtualPageView is still supposed to use the old system, which I have removed.

File: src/utils/consent.js [hunk 4]
```diff
@@ -8,10 +8,14 @@ export const isConsentGranted = () => {
...
-    app_id: appInstallId()
-  })
+  sendEvent(
+    '/analytics/legacy/kaiosappfirstrun/1.0.0',
+    'eventlogging_KaiOSAppFirstRun', 
+    {
+      app_version: version,
+      app_id: appInstallId()
+    }
+)
```
Comment (hueitan) [inline src/utils/consent.js:18]: my bad, should be like this

```suggestion
    {
      app_version: version,
      app_id: appInstallId()
    }
)
```
```suggestion
  sendEvent(
    '/analytics/legacy/kaiosappfirstrun/1.0.0',
    'eventlogging_KaiOSAppFirstRun', 
    {
      app_version: version,
      app_id: appInstallId()
    }
  )
```

File: src/utils/sendEvent.js [hunk 5]
```diff
@@ -1,22 +1,12 @@
...
-    revision,
-    event,
-    webHost: `${language}.wikipedia.org`,
-    wiki: `${language}wiki`
-  })
-  if (isUrlValid(url)) {
-    navigator.sendBeacon(url)
-  }
+export const sendEvent = ($schema, stream, event) => {
+  const body = JSON.stringify([{
```
[the remark this answers is not in the document]
Comment (hueitan) [reply src/utils/sendEvent.js:6]: or does it mean we can send multiple events? [ event1{}, event2{} ]

[the remark this answers is not in the document]
Comment (stephanebisson) [reply src/utils/sendEvent.js:6]: Yes, we can send multiple events but for now I'm going to remove those brackets so it is more clear that we are sending a single object.

**Comment by stephanebisson** [comment]
I'm going to merge this now so it starts sending event to the new system so we can test properly.

**Pull request merged**